packages/vite/src/node/config.ts

```diff
@@ -612,8 +653,18 @@ export async function resolveConfig(
     getSortedPluginHooks: undefined!
   }
 
+  const resolvedConfigFile = configFile ? normalizePath(configFile) : undefined
+
+  const resolvedServerPersistentCacheOptions =
+    await resolvePersistentCacheOptions({
+      config,
+      pkgPath,
+      resolvedRoot,
+      resolvedConfigFile
```

Comment:
I think the hash version should contains the hash of the whole bundled config instead of just the entry point. 

Here is how I did it when I tried to implement this feature: https://github.com/vitejs/vite/pull/4120/files#diff-11e17761d4ecfee8f8fde15c6d79b7bc0260176396a30dfd8e6f6bbaf5af4745